check/configuration/custom-linters/reference.md

```diff
@@ -0,0 +1,527 @@
+
+### `good_without_config`
+
+`good_without_config`: *optional boolean*. Indicates whether this linter is recommended without 
+the user tuning its configuration. Prefer `suggest_if`.
+
+### `files`
+
+`files` is a list of file types listed in the `lint.files` section that this linter applies to. 
+
+
+### `include_lfs`
+
+`include_lfs`: boolean, exclude this filetype if it is tracked using LFS.
+
+[//]: # (`std_in`: boolean, Should the command be fed the file on standard input?)
+
+[//]: # (`cache_results` : boolean, indicates whether or not to support caching for this linter.)
+
+[//]: # (`disable_upstream` :boolean, indicates whether or not we support comparing against the upstream version of this file.  **should we expose this?**)
+
+### `symlinks`
+
+`symlinks`: a list of symlinks to be created when using sandboxing.
+
+### `environment`
+
+`environment`: a list of runtime variables used when running the linter.
+
+### `is_recommended`
+
+`is_recommended`: *boolean*. Indicating whether `trunk init` should try to enable this linter. 
+Prefer [`suggest_if`](#suggest_if)
+
+### `run_linter_from`
+
+`run_linter_from`: indicates whether to set current working directory to WORKSPACE root, or the 
+target files folder when run. Prefer `run_from` at the command level.
+
+`include_scanner_type`: which include scanner to use, if any.
+
+### `formatter`
+
+`formatter`: boolean. Indicates whether this is a formatter and should be included in `trunk fmt`
+
+### `allow_empty_files`
+
+`allow_empty_files`: boolean. Indicates to skip linting empty files for this linter.
+
+### `runtime`
+
+`runtime`: RuntimeType, Which runtime, if any, to require to be setup for this linter.
+
+### `package`
+
+`package`: string, What primary package to install, if using a package manager runtime
+
+### `extra_packages`
+
+`extra_packages`: list of strings, Extra packages to install, versions are optional
+
+### `download`
+
+`download`: *string*. The download url. You must provide either runtime + packages or download, not both. Using runtimes is preferred. [See Runtimes](../../advanced-setup/runtimes.md)
+
+`issue_url_format`: string, a format string that accepts issue codes for links to issues docs.
+
+### `run_from_root_target`
+
+`run_from_root_target`: *string*. Walk up to find this file to detect the run from directory. 
+Prefer `run_from` at the command level.
+
+### `hold_the_line`
+
+`hold_the_line`: *optional boolean*. Whether hold-the-line will be done for this linter or not.
+
+### `read_output_from`
+
+`read_output_from`,  Tell the parser where to expect output from for reading (stdout, stderr, 
+tmp file). [See Output Sources](output-types.md#output-sources)
+
+### `prepare_command`
+
+`prepare_command`, ex. `[tflint, --init]`
+
+[//]: # (`version`, **???** )
+
+### `known_good_version`
+
+`known_good_version`: The version `trunk init` inits with. If not present Trunk will query the 
```

Comment:
This is not quite correct. This field will only be used for tools where trunk does not know how to automatically query the latest version. Currently, trunk can query the latest version for all package managers and downloads hosted on github.